specification/monitor/resource-manager/Microsoft.Insights/stable/2023-10-01/metricDefinitions_API.json

```diff
@@ -0,0 +1,384 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Monitor Metrics Definitions API",
+    "description": "Provides APIs for getting the metric metadata for Azure resources.",
+    "version": "2023-10-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Insights/metricDefinitions": {
+      "get": {
+        "tags": [
+          "MetricDefinitions"
+        ],
+        "operationId": "MetricDefinitions_ListAtSubscriptionScope",
+        "description": "Lists the metric definitions for the subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/RegionParameter"
+          },
+          {
+            "$ref": "./commonMonitoringTypes.json#/parameters/MetricNamespaceParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./commonMonitoringTypes.json#/definitions/ErrorContract"
+            }
+          },
+          "200": {
+            "description": "Successful request to get the list of metric definitions",
+            "schema": {
+              "$ref": "#/definitions/SubscriptionScopeMetricDefinitionCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Get Subscription level Metric Definitions without filter": {
+            "$ref": "./examples/GetMultiResourceMetricDefinitions.json"
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.Insights/metricDefinitions": {
```

Comment:
>/{resourceUri}/providers/Microsoft.Insights/metricDefinitions [](http://example.com/codeflow?start=5&length=61)

is this intended to be an extension resource ? If so is it extending any Azure namespace?